core/src/main/java/apoc/cypher/CypherInitializer.java

```diff
@@ -62,16 +61,16 @@ public void available() {
                     awaitApocProceduresRegistered();
                 }
 
-                var defaultDb = dependencyResolver.resolveDependency(DefaultDatabaseResolver.class).defaultDatabase(null);
-                if (defaultDb.equals(db.databaseName())) {
+                if (isSystemDatabase) {
                     try {
+                        awaitDbmsComponentsProcedureRegistered();
                         final List<String> versions = db.executeTransactionally("CALL dbms.components", Collections.emptyMap(),
                                 r -> (List<String>) r.next().get("versions"));
-                        final String apocFullVersion = Version.class.getPackage().getImplementationVersion();
-                        if (isVersionDifferent(versions, apocFullVersion)) {
+                        final String apocVersion = Version.class.getPackage().getImplementationVersion();
+                        if (isVersionDifferent(versions, apocVersion)) {
                             userLog.warn("The apoc version (%s) and the Neo4j DBMS versions %s are incompatible. \n" +
-                                            "See the compatibility matrix in https://neo4j.com/labs/apoc/4.4/installation/ to see the correct version",
-                                    apocFullVersion, versions.toString());
+                                            "The two first numbers of both versions needs to be the same.",
```

Comment:
Is this true? I thought after 5.0 we wanted to have a compatibility layer that works differently

Reply:
Me, Nacho and Daniel discussed this a bit at the lunch lecture on GDS compatibility layer last week. It is leaning towards that we will not do a compatibility layer, but just say you need to use  e.g. APOC 5.2.0 with Neo4j 5.2.0 or Neo4j 5.2.1 (if there is a security patch)

Reply:
okay :)